codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocket08FrameDecoder.java

```diff
@@ -421,9 +407,14 @@ private void unmask(ByteBuf frame) {
             i += 4;
         }
 
+        //  WebSocketUtil.byteAtIndex assumes BE encoded mask
+        if (order == ByteOrder.LITTLE_ENDIAN) {
+            intMask = Integer.reverseBytes(intMask);
+        }
```

Comment:
Dont we also need to do this with the longMask before we use it in the for loop above ?

Reply:
It seems we are not (as far as my understanding of ByteBuffer get / getLE / order interaction goes) as both mask and payload longs / ints have same endianess. Mask needs to be converted from LE for per-byte unmasking only.